scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/JgitServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import org.eclipse.jgit.lib.Repository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.scriptoocloud.core.exceptions.InvalidGitUrlException;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+import edu.asu.diging.scriptoocloud.core.service.JgitService;
+
+
+/*
+ * Clones remote git repositories to file system utilizing the JGit dependency
+ * 
+ * @author Jason Ormsby
+*/
+
+@Service
+class JgitServiceImpl implements JgitService{
+
+    @Autowired
+    private DeleteFilesService deleteFilesService;
+
+    /*
+     * Creates folder in file system and clones a remote git repository to it
+     * 
+     * @param   localRepoFolderName         Name for new folder that will store cloned remote git repository 
+     * @param   remoteGitRepoUrl            nonmalformed Url passed by user that points to remote git repository
+     * @throws  InvalidGitUrlException      If remote repository doesn't exist 
+     *                                      or jgit encounters failure related to copying remote repository at url
+    */
+    @Override
+    public void clone(String localRepoFolderName, String remoteGitRepoUrl) throws InvalidGitUrlException{
+        try{
+            Git.cloneRepository().setURI(remoteGitRepoUrl).setDirectory(new File(localRepoFolderName)).call().close();
+        } catch(GitAPIException e){
+            deleteFilesService.deleteDirectoryContents(new File(localRepoFolderName));
+            throw new InvalidGitUrlException(e);
+        } catch(JGitInternalException e){
+            deleteFilesService.deleteDirectoryContents(new File(localRepoFolderName));
+            throw new InvalidGitUrlException(e);
```

Comment:
hm, I would rename this exception or use a different one. The exceptions thrown by jgit can be for many reasons not just invalid urls, can they? Also, if you handle both exceptions the same way, you can catch them in the same block `GitAPIException|JGitInternalException e`

Reply:
E.g. I forgot to set up the repositories path, and when I tried to clone a repo I got an invalid url error message which didn't make sense.